src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/ptv/PrintableFormServiceChannelInTestsIT.java

```diff
@@ -0,0 +1,188 @@
+package fi.otavanopisto.kuntaapi.test.server.integration.ptv;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.jayway.restassured.http.ContentType;
+
+import fi.metatavu.kuntaapi.server.rest.model.PrintableFormServiceChannel;
+import fi.metatavu.kuntaapi.server.rest.model.ServiceChannelAttachment;
+import fi.metatavu.ptv.client.model.V7VmOpenApiAddressDeliveryIn;
+import fi.metatavu.ptv.client.model.V7VmOpenApiPrintableFormChannel;
+import fi.metatavu.ptv.client.model.V7VmOpenApiPrintableFormChannelInBase;
+import fi.metatavu.ptv.client.model.VmOpenApiAddressPostOfficeBoxIn;
+import fi.metatavu.ptv.client.model.VmOpenApiAddressStreetIn;
+import fi.metatavu.ptv.client.model.VmOpenApiAttachment;
+import fi.metatavu.ptv.client.model.VmOpenApiLanguageItem;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.translation.PtvAddressSubtype;
+import fi.otavanopisto.kuntaapi.server.persistence.model.clients.AccessType;
+import fi.otavanopisto.kuntaapi.server.persistence.model.clients.ClientOrganizationPermission;
+import fi.otavanopisto.kuntaapi.test.AbstractPtvMocker;
+
+public class PrintableFormServiceChannelInTestsIT extends AbstractPtvInTest {
+
+  private static final String SERVICE_CHANNEL_FIND_PATH = "/printableFormServiceChannels/{kuntaApiChannelId}";
+  /**
+   * Starts WireMock
+   */
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(getWireMockPort());
+  
+  @Before
+  public void beforeTest() throws InterruptedException {
+    getPtvOrganizationMocker().mock(TestPtvConsts.ORGANIZATIONS);
+    getPtvServiceMocker().mock(TestPtvConsts.SERVICES);    
+    getPtvServiceChannelMocker().mock(TestPtvConsts.SERVICE_CHANNELS);
+
+    startMocks();
+    
+    waitApiListCount("/organizations", 3);
+    waitApiListCount("/printableFormServiceChannels", TestPtvConsts.PRINTABLE_FORM_SERVICE_CHANNELS.length);
+    waitApiListCount("/services", TestPtvConsts.SERVICES.length);
+  }
+  
+  @Test
+  public void updatePrintableFormServiceChannelUnauthorized() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
+    PrintableFormServiceChannel kuntaApiResource = getPrintableFormServiceChannel(0, TestPtvConsts.PRINTABLE_FORM_SERVICE_CHANNELS.length);
+    
+    givenReadonly()
+      .body(kuntaApiResource)
+      .contentType(ContentType.JSON)
+      .put(SERVICE_CHANNEL_FIND_PATH, kuntaApiResource.getId())
+      .then()
+      .assertThat()
+      .statusCode(401);
+  }
+  
+  @Test
+  public void updatePrintableFormServiceChannelForbidden() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)